src/components/network-area-diagram-viewer/network-area-diagram-viewer.ts

```diff
@@ -1718,4 +1773,101 @@ export class NetworkAreaDiagramViewer {
         const zoom = this.svgDraw?.zoom() ?? 1;
         this.svgDraw?.zoom(0.9 * zoom);
     }
+
+    public saveSvg() {
+        this.addStyle();
+        const userViewBox: DiagramUtils.ViewBox = {
+            x: this.svgDraw?.viewbox().x ?? 0,
+            y: this.svgDraw?.viewbox().y ?? 0,
+            width: this.svgDraw?.viewbox().width ?? 0,
+            height: this.svgDraw?.viewbox().height ?? 0,
+        };
+        this.zoomToFit();
+        const blobData = [this.getSvg() ?? ''];
+        const blob = new Blob(blobData, { type: 'image/svg+xml' });
+        this.downloadFile(blob, 'nad.svg');
+        this.svgDraw?.viewbox(userViewBox.x, userViewBox.y, userViewBox.width, userViewBox.height);
+        this.removeStyle();
+    }
+
+    private downloadFile(blob: Blob, filename: string) {
+        const a = document.createElement('a');
+        a.download = filename;
+        a.href = URL.createObjectURL(blob);
+        a.click();
+        a.remove();
+    }
+
+    private addStyle() {
+        // add style, if not present
+        if (this.svgParameters.getCssLocation() == CssLocationEnum.EXTERNAL_NO_IMPORT) {
+            const styleElement = DiagramUtils.getStyle(document.styleSheets, this.svgDraw?.node);
+            const gElement = this.svgDraw?.node.querySelector('g');
+            gElement?.before(styleElement);
+        }
+    }
+
+    private removeStyle() {
+        // remove style, if added
+        if (this.svgParameters.getCssLocation() == CssLocationEnum.EXTERNAL_NO_IMPORT) {
+            const styleElement: HTMLElement | null = this.svgDiv.querySelector('style');
+            styleElement?.remove();
+        }
+    }
+
+    public savePng(backgroundColor?: string) {
+        this.copyPng(true, backgroundColor);
+    }
+
+    public screenshot(backgroundColor?: string) {
+        this.copyPng(false, backgroundColor);
+    }
+
+    private copyPng(copyToFile: boolean, backgroundColor?: string) {
+        this.addStyle();
+        this.addBackgroundColor(backgroundColor);
+        const svgXml = DiagramUtils.getSvgXml(this.getSvg());
+        const image = new Image();
+        image.src = svgXml;
+        image.onload = () => {
+            const png = DiagramUtils.getPngFromImage(image);
+            const blob = DiagramUtils.getBlobFromPng(png);
+            if (copyToFile) {
+                this.downloadFile(blob, 'nad.png');
```

Comment:
the downloaded names could be improved, I think we should rather call it "screenshot" + get the network id from the metadata + "_nad.png", and for svg "export" + id + "_nad.svg". Problem is, the metadata does not contain any network id... Let's keep it like this for now.